fairscale/optim/oss.py

```diff
@@ -591,13 +591,14 @@ def _setup_flat_buffers(self) -> None:
 
                     # Merge all the trainable params in a single bucket
                     trainable_params = list(filter(lambda x: x.requires_grad, params))
-                    buffer_size = sum(map(lambda x: x.numel(), trainable_params))
-                    bucket = ParamBucket(size=buffer_size, dtype=params[0].dtype, device=device)
+                    if trainable_params:
+                        buffer_size = sum(map(lambda x: x.numel(), trainable_params))
+                        bucket = ParamBucket(size=buffer_size, dtype=trainable_params[0].dtype, device=device)
```

Comment:
there was a typo here, which could change everything: pick the first param type vs. first trainable param. The trainable params are the ones which go in a bucket, so you could have a config (see added unit test) which could break that, by adding another param of another type which was not trainable, and have it come first in the list